### PR TITLE
ZEP-343 add webhooks

### PR DESCRIPTION
**Pull request by corroded**
## WHAT 

Added webhooks end point:

image

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -896,7 +896,7 @@ class MainClass {
 
 **Step 1. Extract the timestamp and signatures from the header**
 
-Zepto the header, using the `.` (dot) character as the separator, to get a list of elements.
+Split the header, using the `.` (dot) character as the separator, to get a list of elements.
```
Comment (tomdalling) [inline source/index.html.md:901]:
```
class String
  alias_method :zepto, :split
end
```

File: source/openapi3/split.yaml [hunk 2]
```diff
@@ -5641,5 +5789,97 @@ components:
...
+              description: The Webhook Delivery ID
+            webhook_id:
+              type: string
+              format: uuid
+              description: The Webhook ID
+            state:
+              type: string
+              description: The state of the webhook delivery.
+              enum:
+                - pending
```
Comment (tomdalling) [inline source/openapi3/split.yaml:5873]: I believe these are translated into more merchant-friendly terms, for the API.

https://github.com/zeptofs/split/blob/00931172e1fd9ce658112fb378b4879ea6e13dc5/app/decorators/webhook_delivery_decorator.rb#L5-L8

File: source/openapi3/split.yaml [hunk 3]
```diff
@@ -5641,5 +5789,97 @@ components:
...
+                - delivered
+                - will_retry
+                - gave_up
+            payload:
+              type: object
+              description: Could be anything
```
Comment (tomdalling) [inline source/openapi3/split.yaml:5879]:
```suggestion
              description: The webhook delivery's request body. This varies depending on the event type.
```

File: source/openapi3/split.yaml [hunk 4]
```diff
@@ -3290,6 +3290,154 @@ paths:
...
+                $ref: '#/components/schemas/ListAllWebhooksResponse'
+        '404':
+          description: Not Found
+
+  '/webhooks/{webhook_id}/deliveries':
+    get:
+      tags:
+        - Webhooks
+      summary: List deliveries for a webhook
+      description: 'NOTE: There is a historical limit of 30 days for webhook deliveries.'
```
Comment (tomdalling) [inline source/openapi3/split.yaml]: "historical limit" kind of implies that there used to be a limit, but there might not be one any more.

```suggestion
      description: 'NOTE: This is limited to webhook deliveries created in the last 30 days.'
```

File: source/openapi3/split.yaml [hunk 5]
```diff
@@ -3290,6 +3290,154 @@ paths:
...
+            enum: [2xx, 4xx, 5xx]
+          example: '418'
+        - name: state
+          in: query
+          description: 'Filter deliveries by state. NOTE: Delivered means it arrived in the server, i.e. the response code could be 2xx, 4xx, or, 5xx. Single value, exact match'
+          required: false
+          style: simple
+          schema:
+            type: string
+            enum: [pending, delivered, will_retry, gave_up]
```
Comment (tomdalling) [inline source/openapi3/split.yaml]: These should be the public-facing statuses too.

**Review by tomdalling — APPROVED** [review]

File: source/openapi3/split.yaml (hunk 2 above)
Comment (corroded) [reply source/openapi3/split.yaml:5873]: Good catch!

**Pull request merged**